subjects/corewar/README.md

```diff
@@ -0,0 +1,366 @@
+000008a7
+```
+
+- All numbers are represented in hexadecimal.
+- At the start of each line there is the position of the first byte of the line in hexadecimal.
+- This means that for the second line `10` hex is actually 16 in decimal notation.
+- The right panel shows a `.` if the byte is not a valid ASCII character.
+- Repeated lines are aggregated under `*`, this is why we pass from `10` to `80`.
+- Each `xx` is a byte (a byte is 8 bits, `ff` in hexadecimal is `1111 1111` in binary, `255` in decimal).
+
+Let's read it through:
+
+- The first 4 bytes `00 ea 83 f3` are the program signature (also called the magic), this is a 32 bits integer, represented as four 8 bits slices.
+- Then the next 128 bits are for the name of the program, the first five `61 6d 65 62 61` are the actual name, the others are zeros because we don't actually need it.
+- Then we have 4 bytes `00 00 00 17` (so 23) which will be a 32 bits integer with the size in bytes of the program to be executed in the arena of the VM (so only the size of the instructions, without the name, description, signature and so on).
+- Then we have the description which works exactly the same as the name but it will adds four bytes padding at the end of it.
+
+The actual instructions starts from there:
+
+- The instruction `sti` is made by those bytes: `0b 68 01 00 0f 00 01`.
+- `0b` is the opcode, `68` is the pcode, `01` is the first parameter, `00 0f` is the second parameter and `00 01` is the third parameter.
+- Notice the last two parameters are written on two bytes each because `sti` has `Has Idx` set to true.
+- Next instruction is `and`, and it is `06 64 01 00 00 00 00 01`.
+- `06` is the opcode, `64` is the pcode, the first parameter is `01`, the second one is `00 00 00 00` and the third one is `01`.
+- Here comes the `live` instruction: `01 00 00 00 01`.
+- `01` is the opcode, `00 00 00 01` is the first parameter, notice the parameter memory space will be overridden by the `sti` instruction.
+- The last instruction is `zjmp`: `09 ff fb`.
+- `01` is the opcode while `ff fb` is the first parameter (on two bytes because of `Has Idx` once again).
+
+#### Testing environment
+
+You will be provided with a [playground](data/playground.zip) containing:
+
+- A VM and an Assembler binaries to use as references.
+- A Dockerfile to run the binaries in a container.
+- Instructions on how to execute the references.
+- Some players to use as tests.
+
+> The provided VM has an extra flag `-v` which you can use to print the state of the VM at every cycle, this should greatly help you during development and debugging.
+
+### Some advices for the road
+
+While the functioning of the VM and the Assembler may seems quite strange and obscure, it is a perfect scenario to reproduce a CPU and understand what a computer really does at its core.
+
+In this project you are indeed creating a Turing Complete machine largely inspired by Von Neumann architecture (which is still how today's processors works).
```

Comment:
Maybe add some links to wikipedia pages:
- https://en.wikipedia.org/wiki/Turing_machine
- https://en.wikipedia.org/wiki/Von_Neumann_architecture

Reply:
done

Reply:
normally they already are hyperlinks